src/player.cpp

```diff
@@ -2342,6 +2396,268 @@ std::string dodge_skill_text(double mod)
+                                          bool both_sides) const
+{
+    constexpr double torso_multiplier = -1 / 10.0f;
+    constexpr double leg_multiplier = -1 / 20.0f;
+    return calc_flexible_encumbrance( bp_torso, bp_leg_l, torso_multiplier,
+                                      leg_multiplier, total, encumbered_part, both_sides );
+}
+
+void player::dodge_encumbrance_text( body_part encumbered_part, std::string &s,
+                                    bool both_sides ) const
+{
+    double dodge_penalty = dodge_encumbrance( false, encumbered_part, both_sides );
+    s += string_format( _( "Dodge penalty %+.2f. "), dodge_penalty );
+}
+
+void player::fall_damage_encumbrance_text( body_part encumbered_part, std::string &s,
+                                    bool both_sides ) const
+{
+    float fall_damage_penalty = ( fall_damage_mod() -
+                                  fall_damage_encumbrance( true, encumbered_part, both_sides) ) * 100.0f;
+    s += string_format( _( "Falling damage %+.1f percentage points. "), fall_damage_penalty );
+}
+
+void player::move_cost_encumbrance_text ( body_part encumbered_part, std::string &s,
+                                          bool both_sides ) const
+{
+    constexpr int base_run_cost = 100;
+    bool diagonal = false;
+    bool exclude = true;
+    int move_cost = ( run_cost( base_run_cost, diagonal ) -
+                      move_cost_encumbrance( base_run_cost, diagonal, exclude, encumbered_part,
+                                             both_sides) );
+    s += string_format( ngettext( "Walking or running costs %+d movement point. ",
+                                  "Walking or running costs %+d movement points. ",
+                                  move_cost ), move_cost );
+}
+
+void player::swim_cost_encumbrance_text( body_part encumbered_part, std::string &s,
+                                        bool both_sides ) const
+{
+    bool exclude_encumbered = false;
+    bool exclude_unencumbered = true;
+    bool partial_exclude = true;
+    int swim_cost = ( swim_speed_encumbrance( exclude_encumbered, partial_exclude, encumbered_part,
+                                              both_sides) -
+                      swim_speed_encumbrance( exclude_unencumbered, partial_exclude, encumbered_part,
+                                              both_sides) );
+    s += string_format( ngettext( "Swimming costs %+d movement point. ",
+                                  "Swimming costs %+d movement points. ",
+                                  ( swim_cost == 0 ? 2 : swim_cost ) ), swim_cost );
+}
+
+void player::exclude_body_parts( int &encumb1, const body_part part1, int &encumb2,
+                                 const body_part part2, int &encumb3,const body_part part3,
+                                 const bool exclude, const body_part excluded_part,
+                                 const bool both_sides ) const
```

Comment:
It does have a very short comment in the header, but I'll add a proper description of the function and all its parameters to that.